pkg/services/sso/redhatsso_service.go

```diff
@@ -159,7 +159,7 @@ func (r *redhatssoService) GetKafkaClientSecret(accessToken string, clientId str
 	if err != nil {
 		return "", errors.NewWithCause(errors.ErrorFailedToGetSSOClientSecret, err, "failed to get sso client secret")
 	}
-	if !found {
+	if found {
 		//if client is found re-generate the client secret.
 		svcData, seErr := r.client.RegenerateClientSecret(accessToken, shared.SafeString(serviceAccount.Id))
```

Comment:
Is this correct? Do we regenerate the secret everytime we get it? @ziccardi @akoserwal

Reply:
@JameelB My memory can be wrong, but I believe the reason behind that was that on redhatsso the only way to access the secret is by regenerating it (something similar to what happens with github tokens), so we decided that, for the moment, we always regenerate it. However I think it can be removed now. I leave the final word to @akoserwal

Reply:
Don't think we use this in the internal kafka services but I see it being used in connectors still: https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/main/internal/connector/internal/services/connector_cluster.go#L936

Reply:
Yeah, the method name doesn't make it obvious that the secret is actually regenerated as there is no way to retrieve it in redhat sso. That's what we are using it for in connectors.

Reply:
make sense.